src/main/java/core/basesyntax/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+
+    private final Storage storage;
+
+    public ReportGeneratorImpl(Storage storage) {
```

Comment:
This class should not be dependent on the Storage class. It's better to pass the data to the `getReport()` method as a parameter. This way, the class will be more flexible and can generate a report from any data source.